src/mp4box/stsd.rs

```diff
@@ -40,23 +40,26 @@ pub struct StsdBox {
     /// TTXT subtitle codec
     #[serde(skip_serializing_if = "Option::is_none")]
     pub tx3g: Option<Tx3gBox>,
+
+    /// Unrecognized codecs
+    pub unknown: Vec<FourCC>,
 }
 
 impl StsdBox {
-    pub fn kind(&self) -> TrackKind {
+    pub fn kind(&self) -> Option<TrackKind> {
         if self.av01.is_some()
             || self.avc1.is_some()
             || self.hvc1.is_some()
             || self.vp08.is_some()
             || self.vp09.is_some()
         {
-            TrackKind::Video
+            Some(TrackKind::Video)
         } else if self.mp4a.is_some() {
-            TrackKind::Audio
+            Some(TrackKind::Audio)
         } else if self.tx3g.is_some() {
-            TrackKind::Subtitle
+            Some(TrackKind::Subtitle)
         } else {
-            TrackKind::Video
+            None
```

Comment:
Wouldn't it make more sense to add `TrackKind::Unknown`?

Reply:
Imho `Option` here makes it a little more explicit that this is something that should be handled on a different path from having a `TrackKind`. In general I don't like adding "invalid" or "error" cases to enums if possible, to preserve the "fully parsed and thus valid" property of the type.

Reply:
imho `None` is lying a little bit since it's not no track type it's just a track type we didn't detect properly. But the argument of  "a TrackKind is always a well parsed & well behaved track kind" makes sense to me :)